vertx-web-client/src/main/java/io/vertx/ext/web/client/impl/HttpContext.java

```diff
@@ -443,11 +443,14 @@ private void handleCreateRequest() {
           fail(e);
           return;
         }
-        for (String headerName : this.request.headers().names()) {
-          requestOptions.putHeader(headerName, this.request.headers().get(headerName));
+        MultiMap headers = this.request.checkHeaders(requestOptions);
+        MultiMap requestHeaders = this.request.headers();
+        for (String headerName : requestHeaders.names()) {
+          headers.remove(headerName);
         }
+        headers.addAll(requestHeaders);
```

Comment:
Actually, I do not quite understand why headers are copied twice. See https://github.com/vert-x3/vertx-web/blob/2a661ad6d4e163af5bd4e9a0ee3cf5609382356b/vertx-web-client/src/main/java/io/vertx/ext/web/client/impl/HttpRequestImpl.java#L538)

Without the second copying all tests are also passed. Check test commit https://github.com/ilia1243/vertx-web/commit/f470308fc5ed5598d15f96ef8c4cfcb270f1f0de

Though I see some activity with this code
1. https://github.com/vert-x3/vertx-web/commit/8c482962
2. https://github.com/vert-x3/vertx-web/commit/4ea2fb35

I did not remove this part, because do not see the whole picture. Please comment if this can be removed.